src/modelinstance.cpp

```diff
@@ -1167,6 +1167,58 @@ Status ModelInstance::infer(const tensorflow::serving::PredictRequest* requestPr
     return StatusCode::OK;
 }
 
+Status ModelInstance::infer(const ::inference::ModelInferRequest* requestProto,
```

Comment:
Could infer method be templatized? I think we would only need few specifications like getModelName from proto and getVersion from proto?

Reply:
This method is virtual so we cannot templatize it. Plan is to do it when we do stateful for KFS. Then with refactor there we should be able to do it.